src/openms/source/QC/Contaminants.cpp

```diff
@@ -35,113 +35,181 @@
 #include <OpenMS/QC/Contaminants.h>
 #include <include/OpenMS/CHEMISTRY/ProteaseDigestion.h>
 #include <include/OpenMS/METADATA/ProteinIdentification.h>
-#include <include/OpenMS/FORMAT/FeatureXMLFile.h>
-//#include <include/OpenMS/CHEMISTRY/DigestionEnzymeDB.h>
-//#include <include/OpenMS/CHEMISTRY/ProteaseDB.h>
 
-using namespace OpenMS;
-using namespace std;
 
+using namespace std;
 
 
-void Contaminants::compute(FeatureMap& features, const std::vector<FASTAFile::FASTAEntry>& contaminants)
+namespace OpenMS
 {
-  if (digested_db_.empty())
+
+  void Contaminants::compute(FeatureMap &features, const std::vector<FASTAFile::FASTAEntry> &contaminants)
   {
-    String enzyme = features.getProteinIdentifications()[0].getSearchParameters().digestion_enzyme.getName();
-    if (enzyme == "unknown_enzyme")
+    //empty FeatureMap
+    if (features.empty())
     {
-      LOG_WARN << "No digestion enzyme in featureMap detected.";
+      LOG_WARN << "FeatureMap is empty." << "\n";
+      return;
     }
-    ProteaseDigestion digestor;
-    digestor.setEnzyme(enzyme);
-    UInt missed_cleavages (features.getProteinIdentifications()[0].getSearchParameters().missed_cleavages);
-    digestor.setMissedCleavages(missed_cleavages);
-
-    Size dropped_by_length(0); // stats for removing candidates
-
-
-    for (const FASTAFile::FASTAEntry& fe : contaminants)
+    //empty features
+    if (features[0].getPeptideIdentifications().empty())
```

Comment:
remove this check ... it only checks if the first feature has pepIds... it may or may not ...